src/sim.rs

```diff
@@ -48,15 +48,24 @@ impl Sim {
             world.register(addr, epoch, notify.clone());
         }
 
-        let handle = World::enter(&self.world, || rt.with(|| tokio::task::spawn_local(host)));
+        let handle = World::enter(&self.world, || rt.with(|| tokio::task::spawn_local(client)));
 
         self.rts.insert(addr, Role::client(rt, handle));
     }
 
     /// Register a host with the simulation
-    pub fn host<F>(&mut self, addr: impl ToSocketAddr, host: F)
+    ///
+    /// This method takes a `Fn` that builds a future, as opposed to
+    /// [`Sim::client`] which just takes a future. The reason for this is we
+    /// might restart the host, and so need to be able to call the future
+    /// multiple times.
+    ///
+    // TODO: Removing the `'static` bound on `F` would make writing some tests
```

Comment:
I'm not seeing how we would do this due to constraints on `spawn_local`. Did you see a path?

Reply:
I'm not seeing why `spawn_local` is a problem!

The main issue is that the factory fn needs to live somewhere. I've moved it into the new fn that lives on the heap. That works because it has no non-static refs.

I don't know what else could possibly work.